<a href="https://colab.research.google.com/github/dsdayat/Assignment-DS-Day-7-Dibimbing/blob/main/HW_HTnc_Agus_Nur_Hidayat.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Hypothesis testing**

Tugas nya adalah lakukan pengujian hipotesis untuk
membuktikan apakah education berpengaruh terhadap
income atau tidak?



In [56]:
import pandas as pd
import numpy as np
import scipy.stats as st
import seaborn as sns

In [57]:
df=pd.read_csv('marketing_campaign.csv',sep="\t")

In [58]:
df.shape


(2240, 29)

In [59]:
#cek missing value kalo ada, nembak pake median karena aku gatau harus diapain.
df.isnull().sum() 

ID                      0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64

In [60]:
nembakmedian=df['Income'].median()
df['Income']=df['Income'].fillna(nembakmedian)
df.isnull().sum() #cek lagi ada yang missing value apa


ID                     0
Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Dt_Customer            0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Complain               0
Z_CostContact          0
Z_Revenue              0
Response               0
dtype: int64

In [61]:
df.head()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  Teenhome  \
0  5524        1957  Graduation         Single  58138.0        0         0   
1  2174        1954  Graduation         Single  46344.0        1         1   
2  4141        1965  Graduation       Together  71613.0        0         0   
3  6182        1984  Graduation       Together  26646.0        1         0   
4  5324        1981         PhD        Married  58293.0        1         0   

  Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0  04-09-2012       58       635  ...                  7             0   
1  08-03-2014       38        11  ...                  5             0   
2  21-08-2013       26       426  ...                  4             0   
3  10-02-2014       26        11  ...                  6             0   
4  19-01-2014       94       173  ...                  5             0   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0             0             0             0             0         0   
1             0             0             0             0         0   
2             0             0             0             0         0   
3             0             0             0             0         0   
4             0             0             0             0         0   

   Z_CostContact  Z_Revenue  Response  
0              3         11         1  
1              3         11         0  
2              3         11         0  
3              3         11         0  
4              3         11         0  

[5 rows x 29 columns]

In [62]:
df.tail()

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
2235  10870        1967  Graduation        Married  61223.0        0   
2236   4001        1946         PhD       Together  64014.0        2   
2237   7270        1981  Graduation       Divorced  56981.0        0   
2238   8235        1956      Master       Together  69245.0        0   
2239   9405        1954         PhD        Married  52869.0        1   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
2235         1  13-06-2013       46       709  ...                  5   
2236         1  10-06-2014       56       406  ...                  7   
2237         0  25-01-2014       91       908  ...                  6   
2238         1  24-01-2014        8       428  ...                  3   
2239         1  15-10-2012       40        84  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[5 rows x 29 columns]

In [63]:
N = 2240 # jumlah populasinya
e = 0.05 # parameter margin of errornya

In [64]:
def slovin(N,e):
    return N/(1+N * e**2)

In [65]:
Ns = slovin(N,e) 
print(Ns)

339.3939393939393


**Sample proportional stratified**

Populasi =2240 Sample dengan sample slovin = 339  . karena kita mau pake proportionate stratified random sampling jadi kita cari tahu jumlah populasi di tiap-tiap kelas variabel bebasnya dengan "value_counts()" atau "df.groupby(['kolom'],as_index=False).count()"
CMIIW



In [66]:
df['Education'].value_counts()

Graduation    1127
PhD            486
Master         370
2n Cycle       203
Basic           54
Name: Education, dtype: int64

In [67]:
df.groupby(['Education'],as_index=False).count()

Education    ID  Year_Birth  Marital_Status  Income  Kidhome  Teenhome  \
0    2n Cycle   203         203             203     203      203       203   
1       Basic    54          54              54      54       54        54   
2  Graduation  1127        1127            1127    1127     1127      1127   
3      Master   370         370             370     370      370       370   
4         PhD   486         486             486     486      486       486   

   Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  AcceptedCmp3  \
0          203      203       203  ...                203           203   
1           54       54        54  ...                 54            54   
2         1127     1127      1127  ...               1127          1127   
3          370      370       370  ...                370           370   
4          486      486       486  ...                486           486   

   AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Complain  \
0           203           203           203           203       203   
1            54            54            54            54        54   
2          1127          1127          1127          1127      1127   
3           370           370           370           370       370   
4           486           486           486           486       486   

   Z_CostContact  Z_Revenue  Response  
0            203        203       203  
1             54         54        54  
2           1127       1127      1127  
3            370        370       370  
4            486        486       486  

[5 rows x 29 columns]

In [68]:
p0=203
p1=54
p2=1127
p3=370
p4=486

In [69]:
#mencari Npengali tiap kelas (Np)
Np = Ns/N
print(Np)

0.1515151515151515


In [70]:
#menentukan jumlah sample tiap kelas
Np0=Np*p0
print ("ntk 0=  ", Np0)
Np1=Np*p1
print ("ntk 1=  ", Np1)
Np2=Np*p2
print ("ntk 2=  ", Np2)
Np3=Np*p3
print ("ntk 3=  ", Np3)
Np4=Np*p4
print ("ntk 4=  ", Np4)

ntk 0=   30.757575757575754
ntk 1=   8.18181818181818
ntk 2=   170.75757575757572
ntk 3=   56.060606060606055
ntk 4=   73.63636363636363


Karena terdapat 5 kelas tingkat pendidikan pada variabel education dengan populasi yang berbeda maka pengambilan sampel dilakukan dengan hasil slovin dibagi hasil kelas 339/5 sebagai Np alias Npembagi. 

Besarnya sampel tiap kelas (Ntk) = Np/p*Ns

dan dapet

ntk 0=   30.757575757575754==> 31

ntk 1=   8.18181818181818==>8

ntk 2=   170.75757575757572==>171

ntk 3=   56.060606060606055==>56

ntk 4=   73.63636363636363==>74



In [71]:
#mendefine kelas education berdasarkan index dan mendklarasi jumlah sampelnya
df0 = df[df['Education'] == '2n Cycle']
df1 = df[df['Education'] == 'Basic']
df2 = df[df['Education'] == 'Graduation']
df3 = df[df['Education'] == 'Master']
df4 = df[df['Education'] == 'PhD']

df0c=df0.sample(31)
df1c=df1.sample(8)
df2c=df2.sample(171)
df3c=df3.sample(56)
df4c=df4.sample(74)


In [72]:
#gabungkan sample tiap kelas ke kedalam sample education, harus sama dengan nilai slovin.
Nedu =pd.concat([df0c, df1c,df2c,df3c,df4c],axis=0)


In [73]:
print(Nedu['Education'])

1644    2n Cycle
1166    2n Cycle
682     2n Cycle
1381    2n Cycle
98      2n Cycle
          ...   
1427         PhD
630          PhD
354          PhD
2165         PhD
1840         PhD
Name: Education, Length: 340, dtype: object


In [74]:
Nedu['Education'].value_counts()#ini detail samplenya

Graduation    171
PhD            74
Master         56
2n Cycle       31
Basic           8
Name: Education, dtype: int64

In [75]:
Nedu.shape#ini total sample sudah didapatkan Nedu, ingat harus sama dengan nilai sample slovin

(340, 29)

**Lanjut ke Anova test**

Karena variabel bebasnya banyak (tipe stratified) makanya pake anova.

Dan karena sudah dapet sample yang dipisah bisa langsung uji anova.

digunakan kriiteria alpha =0.05



H0:df0=df1=df2=df3=df4  (Tidak ada mean yang berbeda dalam kelompok kategori)

H1 = Sedikitnya ada satu mean yang berbeda dalam kelompok kategori

pv<F0 = education tidak berpengaruh terhadap income, H1 ditolak

pv>F0 = education berpengaruh terhadap income, H1 diterima

In [76]:
alp=0.05
anova_test = st.f_oneway(df0c['Income'],
                         df1c['Income'],
                         df2c['Income'],
                         df3c['Income'],
                         df4c['Income'],
                         )

In [77]:
pv= anova_test.pvalue
print(pv)

4.1789569437150215e-06


In [78]:
if (pv<alp):
  print("Education tidak berpengaruh terhadap income")
else:
  print("Education berpengaruh terhadap income")

Education tidak berpengaruh terhadap income


**Statistika Deskriptif**



In [79]:
#cara cepet pake groupby liat meannya ada yang beda apa kaga.
Nedu.groupby(['Education'])['Income'].mean()

Education
2n Cycle      45491.096774
Basic         19220.375000
Graduation    54773.690058
Master        47809.223214
PhD           53840.878378
Name: Income, dtype: float64

cara manual bandingin antara sample pake describe

atau

 manual cari df0c.mean(),df0c.median(), etc.

---



In [80]:
df0c.describe()

ID   Year_Birth        Income    Kidhome   Teenhome  \
count     31.000000    31.000000     31.000000  31.000000  31.000000   
mean    4060.419355  1972.064516  45491.096774   0.419355   0.354839   
std     3003.408395    12.492492  22513.370878   0.501610   0.486373   
min       75.000000  1951.000000   7500.000000   0.000000   0.000000   
25%     1502.000000  1962.500000  26670.000000   0.000000   0.000000   
50%     3865.000000  1975.000000  41039.000000   0.000000   0.000000   
75%     5731.000000  1981.000000  65653.500000   1.000000   1.000000   
max    10401.000000  1995.000000  87000.000000   1.000000   1.000000   

         Recency    MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  31.000000   31.000000   31.000000        31.000000        31.000000   
mean   46.645161  129.193548   22.612903       141.709677        49.258065   
std    30.033923  175.250757   29.554106       215.817700        67.501589   
min    10.000000    0.000000    0.000000         2.000000         0.000000   
25%    20.000000    8.500000    6.000000        11.000000         6.500000   
50%    42.000000   34.000000   16.000000        34.000000        15.000000   
75%    75.000000  201.500000   27.500000       148.500000        73.500000   
max    99.000000  600.000000  162.000000       818.000000       258.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          31.000000     31.000000          31.0          31.0   
mean   ...           5.258065      0.064516           0.0           0.0   
std    ...           2.337630      0.249731           0.0           0.0   
min    ...           1.000000      0.000000           0.0           0.0   
25%    ...           3.000000      0.000000           0.0           0.0   
50%    ...           6.000000      0.000000           0.0           0.0   
75%    ...           7.000000      0.000000           0.0           0.0   
max    ...           9.000000      1.000000           0.0           0.0   

       AcceptedCmp1  AcceptedCmp2   Complain  Z_CostContact  Z_Revenue  \
count     31.000000          31.0  31.000000           31.0       31.0   
mean       0.064516           0.0   0.032258            3.0       11.0   
std        0.249731           0.0   0.179605            0.0        0.0   
min        0.000000           0.0   0.000000            3.0       11.0   
25%        0.000000           0.0   0.000000            3.0       11.0   
50%        0.000000           0.0   0.000000            3.0       11.0   
75%        0.000000           0.0   0.000000            3.0       11.0   
max        1.000000           0.0   1.000000            3.0       11.0   

        Response  
count  31.000000  
mean    0.096774  
std     0.300537  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]

In [81]:
df1c.describe()

ID   Year_Birth        Income   Kidhome  Teenhome   Recency  \
count      8.000000     8.000000      8.000000  8.000000   8.00000   8.00000   
mean    5799.500000  1970.750000  19220.375000  0.375000   0.25000  42.25000   
std     3510.803001    14.488912   6623.206581  0.517549   0.46291  23.74718   
min      880.000000  1947.000000   9722.000000  0.000000   0.00000   7.00000   
25%     3233.500000  1962.500000  15740.000000  0.000000   0.00000  34.00000   
50%     6371.500000  1969.500000  16918.500000  0.000000   0.00000  48.00000   
75%     8478.250000  1980.500000  24037.750000  1.000000   0.25000  51.25000   
max    10250.000000  1992.000000  28389.000000  1.000000   1.00000  80.00000   

       MntWines  MntFruits  MntMeatProducts  MntFishProducts  ...  \
count   8.00000   8.000000         8.000000         8.000000  ...   
mean    2.87500   9.125000         7.750000         6.875000  ...   
std     2.10017   7.298483         5.230406         4.086126  ...   
min     1.00000   3.000000         3.000000         2.000000  ...   
25%     1.00000   3.750000         3.750000         5.000000  ...   
50%     2.50000   7.000000         5.500000         7.000000  ...   
75%     3.75000  11.000000        11.750000         8.000000  ...   
max     6.00000  23.000000        16.000000        15.000000  ...   

       NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count           8.000000      8.000000           8.0           8.0   
mean            6.250000      0.125000           0.0           0.0   
std             1.908627      0.353553           0.0           0.0   
min             3.000000      0.000000           0.0           0.0   
25%             5.500000      0.000000           0.0           0.0   
50%             6.500000      0.000000           0.0           0.0   
75%             8.000000      0.000000           0.0           0.0   
max             8.000000      1.000000           0.0           0.0   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count           8.0           8.0       8.0            8.0        8.0   
mean            0.0           0.0       0.0            3.0       11.0   
std             0.0           0.0       0.0            0.0        0.0   
min             0.0           0.0       0.0            3.0       11.0   
25%             0.0           0.0       0.0            3.0       11.0   
50%             0.0           0.0       0.0            3.0       11.0   
75%             0.0           0.0       0.0            3.0       11.0   
max             0.0           0.0       0.0            3.0       11.0   

       Response  
count  8.000000  
mean   0.125000  
std    0.353553  
min    0.000000  
25%    0.000000  
50%    0.000000  
75%    0.000000  
max    1.000000  

[8 rows x 26 columns]

In [82]:
df2c.describe()

ID   Year_Birth         Income     Kidhome    Teenhome  \
count    171.000000   171.000000     171.000000  171.000000  171.000000   
mean    5487.666667  1968.719298   54773.690058    0.391813    0.532164   
std     3405.107534    11.767484   21250.434608    0.546371    0.545425   
min       55.000000  1945.000000    4861.000000    0.000000    0.000000   
25%     2471.000000  1959.500000   37055.000000    0.000000    0.000000   
50%     5371.000000  1969.000000   56253.000000    0.000000    1.000000   
75%     8707.000000  1978.000000   72224.000000    1.000000    1.000000   
max    11148.000000  1995.000000  105471.000000    2.000000    2.000000   

          Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  171.000000   171.000000  171.000000       171.000000       171.000000   
mean    51.450292   325.040936   39.152047       188.333333        52.046784   
std     29.033482   337.047585   53.075642       219.974410        69.803406   
min      0.000000     2.000000    0.000000         1.000000         0.000000   
25%     28.000000    24.000000    2.000000        15.500000         3.000000   
50%     51.000000   249.000000   11.000000        92.000000        16.000000   
75%     77.000000   522.500000   63.500000       296.000000        72.000000   
max     99.000000  1492.000000  199.000000       940.000000       250.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...         171.000000    171.000000    171.000000    171.000000   
mean   ...           5.017544      0.070175      0.093567      0.099415   
std    ...           2.481635      0.256193      0.292081      0.300097   
min    ...           1.000000      0.000000      0.000000      0.000000   
25%    ...           3.000000      0.000000      0.000000      0.000000   
50%    ...           5.000000      0.000000      0.000000      0.000000   
75%    ...           7.000000      0.000000      0.000000      0.000000   
max    ...          14.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2    Complain  Z_CostContact  Z_Revenue  \
count    171.000000    171.000000  171.000000          171.0      171.0   
mean       0.099415      0.035088    0.029240            3.0       11.0   
std        0.300097      0.184542    0.168973            0.0        0.0   
min        0.000000      0.000000    0.000000            3.0       11.0   
25%        0.000000      0.000000    0.000000            3.0       11.0   
50%        0.000000      0.000000    0.000000            3.0       11.0   
75%        0.000000      0.000000    0.000000            3.0       11.0   
max        1.000000      1.000000    1.000000            3.0       11.0   

         Response  
count  171.000000  
mean     0.122807  
std      0.329179  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      0.000000  
max      1.000000  

[8 rows x 26 columns]

In [83]:
df3c.describe()

ID   Year_Birth        Income   Kidhome   Teenhome  \
count     56.000000    56.000000     56.000000  56.00000  56.000000   
mean    5895.035714  1966.625000  47809.223214   0.50000   0.482143   
std     3179.190983    12.438302  18846.834852   0.53936   0.539059   
min      433.000000  1944.000000  10979.000000   0.00000   0.000000   
25%     3208.500000  1956.000000  34010.500000   0.00000   0.000000   
50%     5931.000000  1967.000000  44143.500000   0.00000   0.000000   
75%     8796.250000  1976.000000  62863.500000   1.00000   1.000000   
max    10913.000000  1992.000000  92859.000000   2.00000   2.000000   

         Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  56.000000    56.000000   56.000000        56.000000        56.000000   
mean   48.107143   271.696429   21.446429       152.464286        33.071429   
std    28.880744   311.198500   35.035525       226.185118        60.325136   
min     2.000000     3.000000    0.000000         2.000000         0.000000   
25%    24.000000    27.750000    1.000000        16.000000         0.000000   
50%    48.000000   128.500000    4.500000        41.500000         5.000000   
75%    71.750000   510.000000   25.750000       152.250000        34.500000   
max    98.000000  1184.000000  138.000000       921.000000       224.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          56.000000     56.000000     56.000000     56.000000   
mean   ...           5.732143      0.071429      0.035714      0.035714   
std    ...           1.921089      0.259870      0.187256      0.187256   
min    ...           1.000000      0.000000      0.000000      0.000000   
25%    ...           5.000000      0.000000      0.000000      0.000000   
50%    ...           6.000000      0.000000      0.000000      0.000000   
75%    ...           7.000000      0.000000      0.000000      0.000000   
max    ...           9.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  \
count     56.000000          56.0      56.0           56.0       56.0   
mean       0.017857           0.0       0.0            3.0       11.0   
std        0.133631           0.0       0.0            0.0        0.0   
min        0.000000           0.0       0.0            3.0       11.0   
25%        0.000000           0.0       0.0            3.0       11.0   
50%        0.000000           0.0       0.0            3.0       11.0   
75%        0.000000           0.0       0.0            3.0       11.0   
max        1.000000           0.0       0.0            3.0       11.0   

        Response  
count  56.000000  
mean    0.089286  
std     0.287736  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]

In [84]:
df4c.describe()

ID   Year_Birth        Income    Kidhome   Teenhome  \
count     74.000000    74.000000     74.000000  74.000000  74.000000   
mean    5343.202703  1965.351351  53840.878378   0.486486   0.702703   
std     3459.794384    11.142210  16466.585800   0.555008   0.613316   
min       17.000000  1943.000000  18988.000000   0.000000   0.000000   
25%     2528.500000  1957.000000  40387.250000   0.000000   0.000000   
50%     5078.000000  1965.500000  52377.000000   0.000000   1.000000   
75%     8898.250000  1973.000000  66267.500000   1.000000   1.000000   
max    11133.000000  1991.000000  86979.000000   2.000000   2.000000   

        Recency     MntWines   MntFruits  MntMeatProducts  MntFishProducts  \
count  74.00000    74.000000   74.000000        74.000000        74.000000   
mean   48.00000   369.986486   16.567568       136.783784        25.337838   
std    27.77108   385.846108   27.551735       190.071167        42.905208   
min     0.00000     6.000000    0.000000         1.000000         0.000000   
25%    22.50000    37.250000    0.000000        16.250000         0.000000   
50%    52.50000   186.500000    2.000000        63.000000         6.000000   
75%    70.75000   664.750000   21.000000       170.500000        28.750000   
max    98.00000  1245.000000  115.000000       899.000000       193.000000   

       ...  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  \
count  ...          74.000000     74.000000     74.000000     74.000000   
mean   ...           5.351351      0.094595      0.108108      0.040541   
std    ...           2.266084      0.294652      0.312636      0.198569   
min    ...           1.000000      0.000000      0.000000      0.000000   
25%    ...           4.000000      0.000000      0.000000      0.000000   
50%    ...           5.000000      0.000000      0.000000      0.000000   
75%    ...           7.000000      0.000000      0.000000      0.000000   
max    ...           9.000000      1.000000      1.000000      1.000000   

       AcceptedCmp1  AcceptedCmp2   Complain  Z_CostContact  Z_Revenue  \
count     74.000000     74.000000  74.000000           74.0       74.0   
mean       0.067568      0.027027   0.013514            3.0       11.0   
std        0.252716      0.163269   0.116248            0.0        0.0   
min        0.000000      0.000000   0.000000            3.0       11.0   
25%        0.000000      0.000000   0.000000            3.0       11.0   
50%        0.000000      0.000000   0.000000            3.0       11.0   
75%        0.000000      0.000000   0.000000            3.0       11.0   
max        1.000000      1.000000   1.000000            3.0       11.0   

        Response  
count  74.000000  
mean    0.148649  
std     0.358170  
min     0.000000  
25%     0.000000  
50%     0.000000  
75%     0.000000  
max     1.000000  

[8 rows x 26 columns]